core/src/main/java/org/elasticsearch/index/fielddata/ScriptDocValues.java

```diff
@@ -585,7 +585,10 @@ public void setNextDocId(int docId) throws IOException {
             if (in.advanceExact(docId)) {
                 resize(in.docValueCount());
                 for (int i = 0; i < count; i++) {
-                    values[i] = in.nextValue();
+                    // We need to make a deepcopy here, because BytesBinaryDVAtomicFieldData's SortedBinaryDocValues
+                    // implementation reuses the returned BytesRef. Otherwise we would end up with the same BytesRef
+                    // instance for all slots in the values array.
+                    values[i] = BytesRef.deepCopyOf(in.nextValue());
```

Comment:
maybe we could also reuse the byte[] like the `ScriptDocValues.Strings` does by keeping an array of `BytesRefBuiler`?